code/modules/html_interface/map/station_map.dm

```diff
@@ -75,8 +92,68 @@
 			var/turf/tile = locate(i, r, map.zCentcomm)
 			if(tile && (is_type_in_list(tile.loc, allowed_areas) && !is_type_in_list(tile.loc, restricted_areas)))
 				if((!istype(tile, /turf/space) && istype(tile.loc, /area/mine/unexplored)) || istype(tile, /turf/simulated/wall) || istype(tile, /turf/unsimulated/mineral) || istype(tile, /turf/unsimulated/wall) || (locate(/obj/structure/grille) in tile) || (locate(/obj/structure/window/full) in tile) || istype(tile, /turf/simulated/shuttle/wall))
-					canvas.DrawBox("#FFFFFFDD", i, r)
+					if(map.holomap_offset_x.len >= map.zCentcomm)
+						canvas.DrawBox("#FFFFFFDD", min(i+map.holomap_offset_x[map.zCentcomm],((2 * world.view + 1)*WORLD_ICON_SIZE)), min(r+map.holomap_offset_y[map.zCentcomm],((2 * world.view + 1)*WORLD_ICON_SIZE)))
+					else
+						canvas.DrawBox("#FFFFFFDD", i, r)
 				else if (istype(tile, /turf/simulated/floor) || istype(tile, /turf/unsimulated/floor) || (locate(/obj/structure/catwalk) in tile) || istype(tile, /turf/simulated/shuttle/floor))
-					canvas.DrawBox("#66666699", i, r)
+					if(map.holomap_offset_x.len >= map.zCentcomm)
+						canvas.DrawBox("#66666699", min(i+map.holomap_offset_x[map.zCentcomm],((2 * world.view + 1)*WORLD_ICON_SIZE)), min(r+map.holomap_offset_y[map.zCentcomm],((2 * world.view + 1)*WORLD_ICON_SIZE)))
+					else
+						canvas.DrawBox("#66666699", i, r)
 
 	centcommMiniMaps[filter] = canvas
+
+/proc/generateStationMinimap(var/StationZLevel)
+	var/icon/canvas = icon('icons/480x480.dmi', "blank")
+
+	for(var/i = 1 to ((2 * world.view + 1)*WORLD_ICON_SIZE))
```

Comment:
This one only does on zLevel specified as argument, and instead of drawing walls and floors, draws area colors.

I guess I should comment my code.